yunwoo/programmers-2/영어-끝말잇기.js

```diff
@@ -0,0 +1,29 @@
+function solution(n, words) {
+  const result = [0, 0];
+  const wordsArr = [words[0]];
+
+  for (let i = 1; i < words.length; i++) {
+    const oneIncrementLength = wordsArr.length + 1;
+    const isLeavingOut =
+      wordsArr.includes(words[i]) ||
+      wordsArr[wordsArr.length - 1].at(-1) !== words[i].at(0);
+
+    if (isLeavingOut) {
+      result[0] = oneIncrementLength % n === 0 ? n : oneIncrementLength % n;
+      result[1] =
+        oneIncrementLength / n === 0 ? n : Math.ceil(oneIncrementLength / n);
+      break;
+    }
+
+    wordsArr.push(words[i]);
+  }
+
+  return result;
+}
```

Comment:
```suggestion
function solution(n, words) {
  const wordsArr = [words[0]];

  for (let i = 1; i < words.length; i++) {
    const oneIncrementLength = wordsArr.length + 1;
    const curWord = words[i];
    const lastWord = wordsArr.at(-1);
    const isLeavingOut =
      wordsArr.includes(curWord) || curWord[0] !== lastWord.at(-1);
    const humanSeq = oneIncrementLength % n || n;
    const turn = Math.ceil(oneIncrementLength / n) || n;

    if (isLeavingOut) return [humanSeq, turn];
    wordsArr.push(words[i]);
  }
  return [0, 0];
}
```
- 풀이잘봤습니다
- 계속 고민해보시면 좋은게 변수를 활용해서 복잡성을 줄이고 가독성을 높일 수 있을지 고민해보시면 좋을 것 같습니다!~
- 삼항 연산자도 좋지만 위 풀이 코드와 같은 경우에는 `||` 연산자를 활용해서도 코드 길이를 많이 줄일 수 있습니다
- 또한 break대신 그냥 return을 해버리시면 result라는 변수가 필요없어집니다 👍